apps/full-site-editing/full-site-editing-plugin/full-site-editing/class-full-site-editing.php

```diff
@@ -350,4 +355,128 @@ public function allow_searching_for_templates() {
 		// Setting this to `public` will allow it to be found in the search endpoint.
 		$post_type->public = true;
 	}
+
+	/**
+	 * This will merge the post content with the post template, modifiying the
+	 * $post parameter.
+	 *
+	 * @param WP_Post $post
+	 */
+	public function merge_template_and_post( $post ) {
+		//bail if not a REST API Request
+		if ( defined( 'REST_REQUEST' ) && ! REST_REQUEST ) {
+			return;
+		}
+
+		// bail if the post type is one of the template post types
+		if ( in_array( get_post_type( $post->ID ), $this->template_post_types ) ) {
+			return;
+		}
+
+		$template_id = get_post_meta( $post->ID, '_wp_template_id', true );
```

Comment:
Once https://github.com/Automattic/wp-calypso/pull/33878/files#diff-ccde406c3871d99ef7cb7fb5df0f7c76R54 is merged we might be able to reuse `A8C_WP_Template` class here, and it's `get_template_id` method or even better `get_template_content`. It will mostly come in handy because it also provides a fallback to default global page template (based on taxonomy), when specific one is not present in `_wp_template_id`.

Reply:
Noted!

Reply:
https://github.com/Automattic/wp-calypso/pull/34108